core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go

```diff
@@ -0,0 +1,496 @@
+package docker_compose_transpiler
+
+import (
+	"errors"
+	"fmt"
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/joho/godotenv"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/port_spec"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/builtin_argument"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_type_constructor"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_types/directory"
+	port_spec_starlark "github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_types/port_spec"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_types/service_config"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"go.starlark.net/starlark"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	cpuToMilliCpuConstant = 1024
+	bytesToMegabytes      = 1024 * 1024
+	float64BitWidth       = 64
+
+	// Look for an environment variables file at the package root, and if present use the values found there
```

Comment:
TODO: handle the case where the .env is not present? Doesn't docker compose have a way to specify the env file to use